SPARCore/GADArticle.m

```diff
@@ -1,104 +1,99 @@
+    if (nextPageToken) {
+        [token setObject:nextPageToken forKey:API_QUERY_PAGE_TOKEN];
+    }
     
-    [GADRemoteModel fetchModelsWithParams:queryURL
-                          queryParameters:@{}
-                         modelTransformer:^(NSData *jsonData){
-                             return [GADArticle articlesFromJSON:jsonData];
+    [super fetchModelsWithParams:queryURL
+                          queryParameters:token
+                         modelTransformer:^(NSArray *jsonArray) {
+                            return [self articlesFromArray:jsonArray];
                          }
-                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
-                            GADArticle *fullArticle = model[0];
-                            self.content = fullArticle.content;
-                            self.authors = fullArticle.authors;
-                            //completion(); --How should we call the completion handler here?
-                        }];
-    
+                        completionHandler:completion];
 }
 
-+ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
++ (NSArray <GADArticle *> *) articlesFromArray: (NSArray *)jsonArray {
     
-    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
-    NSError *error = nil;
-    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
-                                                         options:kNilOptions error:&error];
-    
-    if (error != nil) {
-        NSLog(@"Error parsing JSON.");
-    }
+    NSMutableArray <GADArticle *> *articles = [NSMutableArray new];
     
     for (NSDictionary *element in jsonArray) {
-        //Map fields of element to fields of article
-        GADArticle *article = [[GADArticle alloc] init];
-        //datePublished field is a UNIX Timestamp number - converting to NSDate here
-        int timeStamp = (int)element[API_DATE_PUBLISHED];
-        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
-        article.brief = element[API_BRIEF];
-        article.headerImage = element[API_HEADER_IMAGE];
-        article.publicationId = element[API_PUBLICATION_ID];
-        article.articleId = element[API_ARTICLE_ID];
-        article.title = element[API_TITLE];
-        if (element[API_CONTENT]) {
-            article.content = element[API_CONTENT];
-            article.authors = element[API_AUTHORS];
-        }
+        GADArticle *article = [self articleFromDictionary:element];
         [articles addObject:article];
     }
     return articles;
 }
 
-+ (NSURL *) baseURL {
-    NSURL *queryURL = [NSURL URLWithString:API_HOSTNAME];
-    queryURL = [NSURL URLWithString:API_PREFIX relativeToURL:queryURL];
-    return queryURL;
++ (GADArticle *) articleFromDictionary: (NSDictionary*)dict {
+    //Map fields of element to fields of article
+    GADArticle *article = [[GADArticle alloc] init];
```

Comment:
We should pick either `[Object new]` or `[[Object alloc] init]`. I'm in the `[[Object alloc] init]` camp (because it's more consistent when we have convenience initializers which set default values for properties; e.g. `[[GADArticle alloc] initWithId:(NSString *)identifier title:(NSString *)title]`) but as long as it's consistent I don't really care.